logos/src/logos/capacity/capacity_planner.py

```
@@ -528,22 +604,36 @@ async def _prepare_existing_lane(
-                        action="wake",
-                        provider_id=provider_id,
-                        lane_id=target.lane_id,
-                        model_name=model_name,
-                        reason="Request-time wake for selected sleeping lane",
-                    ),
-                    timeout_seconds=max(timeout_seconds, self.REQUEST_WAKE_TIMEOUT_SECONDS),
+                # Re-check lane state after acquiring lock — a concurrent task
+                # may have already woken this lane while we waited for the lock.
+                current_lanes = self._safe_get_lanes(provider_id)
+                current_target = next(
+                    (l for l in current_lanes if l.lane_id == target.lane_id),
```

Flake8 (7.3.0)

[error] ambiguous variable name 'l'

(E741)